src/main/java/com/zepben/evolve/services/common/translator/BaseCimToProto.kt

```diff
@@ -38,22 +42,39 @@ fun toPb(cim: OrganisationRole, pb: PBOrganisationRole.Builder): PBOrganisationR
     }
 
 /************ IEC61970 CORE ************/
+fun toPb(cim: NameType, pb: PBNameType.Builder): PBNameType.Builder =
+    pb.apply {
+        pb.name = cim.name
+        pb.description = cim.description
+    }
+
+fun toPb(cim: Name, pb: PBName.Builder): PBName.Builder =
+    pb.apply {
+        pb.name = cim.name
+        pb.type = cim.type?.name
```

Comment:
This will become non null when changing to the constructor arg on the name, however when dealing with nulls in proto a runtime exception will be thrown (proto doesn't like nulls). It would becomes
`cim.type?.name?.let { pb.type = it } ?: pb.clearType()`